pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -796,3 +820,23 @@ def _to_abstract_repr(self) -> dict[str, Any]:
         d = super()._to_abstract_repr()
         d["is_virtual"] = True
         return d
+
+    @staticmethod
+    def from_abstract_repr(obj_str: str) -> VirtualDevice:
+        """Deserialize a noise model from an abstract JSON object.
+
+        Args:
+            obj_str (str): the JSON string representing the noise model
+                encoded in the abstract JSON format.
+        """
+        if not isinstance(obj_str, str):
+            raise TypeError(
+                "The serialized VirtualDevice must be given as a string. "
+                f"Instead, got object of type {type(obj_str)}."
+            )
+
+        # Avoids circular imports
+        device = pulser_abstract_repr.deserializer.deserialize_device(obj_str)
+        if isinstance(device, Device):
+            return device.to_virtual()
```

Comment:
I think this is not a bad idea. Let's just add a `Warning:` section in the docstring to document this behaviour